Sources/SwiftParser/Types.swift

```diff
@@ -1057,20 +1057,23 @@ extension Parser {
         return result
       }
 
-      if self.at(.rightSquareBracket) {
-        let (unexpectedBeforeRSquareBracket, rightSquareBracket) = self.expect(.rightSquareBracket)
+      var lookahead = self.lookahead()
+
+      // If we are at a new line, we cannot assume it's an unfinished dictionary or array.
+      if lookahead.currentToken.isAtStartOfLine {
+        return result
+      }
+
+      if let rightSquareBracket = self.consume(if: .rightSquareBracket) {
```

Comment:
This can just be something like the following, I think
```suggestion
      if let rightSquareBracket = self.consume(if: TokenSpec(.rightSquareBracket, allowStartOfLine: false)) {
```